queries/cow_rewards.sql

```diff
@@ -11,18 +11,21 @@ with trade_hashes as (SELECT solver,
                           ORDER BY s.log_index ASC
                           LIMIT 1
                           ) AS settlement ON true
-                      join solver_competitions on settlement.tx_hash = solver_competitions.tx_hash
+                      join solver_competitions
+                          -- This join also eliminates overlapping
+                          -- trades & settlements between barn and prod DB
+                          on settlement.tx_hash = solver_competitions.tx_hash
                       where block_number between {{start_block}} and {{end_block}})
 
 select concat('0x', encode(solver, 'hex'))          as receiver,
        -- Used to compare with Dune (cross referencing that all trades are accounted for in the orderbook)
        count(*) as num_trades,
-       (sum(reward) * pow(10, 18))::numeric::text   as amount,
+       (sum(coalesce(reward, 0)) * pow(10, 18))::numeric::text   as amount,
        '0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB' as token_address
 from trade_hashes
-         -- Inner join because both prod and staging DB index trades and settlements,
--- but the rewards for each environment are contained only in respective databases.
-         inner join order_rewards
+        -- outer join because there are missing records in order rewards:
+        -- e.g. order_uid 0x9CB39E52C3ABD20F38830E70E31E38DEBC12349CA680E7F1DAA4B47B704394D65FCDC32DFC361A32E9D5AB9A384B890C62D0B8AC634D8F38
+         left outer join order_rewards
```

Comment:
There is a slack discussion here regarding this "outer join"

https://cowservices.slack.com/archives/C0440K40U3H/p1666255761713159

Reply:
outer left join is more correct as it is more fault tolerant, in order to have the num_trades correct. But maybe you wanna update the comment as the order seems to be not from a suitable accounting period

Reply:
I will update the test to. Should be October 18